integrations/access/common/app.go

```diff
@@ -479,3 +487,27 @@ func (a *BaseApp) updateMessages(ctx context.Context, reqID string, tag pd.Resol
 
 	return nil
 }
+
+func (a *BaseApp) getResourceNames(ctx context.Context, req types.AccessRequest) ([]string, error) {
+	resourceNames := make([]string, 0, len(req.GetRequestedResourceIDs()))
+	resourcesByCluster := accessrequest.GetResourceIDsByCluster(req)
+
+	for cluster, resources := range resourcesByCluster {
+		resourceDetails, err := accessrequest.GetResourceDetails(ctx, cluster, a.apiClient, resources)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		for _, resource := range resources {
+			resourceID := types.ResourceIDToString(resource)
+			var resourceName string
+			if details, ok := resourceDetails[resourceID]; ok && details.FriendlyName != "" {
+				resourceName = details.FriendlyName
+			} else {
+				resourceName = resourceID
+			}
```

Comment:
This means the resource might just be called `mynode` instead of `cluster/node/mynode` :thinking: 

Should we add some extra information here?
I just did a test and got the following (expected, I would say)
![image](https://github.com/gravitational/teleport/assets/689271/4dfd099d-e699-44fc-9a74-b2b55eda3c4d)

I think pre-pending the resource name with its kind would be ideal :thinking: 
Otherwise, the name might be too generic/duplicated.
I can have a Node called `myapp` and a (web) App called `myapp`
Given that it is not unique, it might cause confusion.